modules/FindBasisUniversal.cmake

```diff
@@ -0,0 +1,176 @@
+#   THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#   DEALINGS IN THE SOFTWARE.
+#
+
+# Vcpkg distributes imgui as a library with a config file, so try that first --
+# but only if IMGUI_DIR wasn't explicitly passed, in which case we'll look
+# there instead
+find_package(basisu CONFIG QUIET)
+if(basisu_FOUND AND NOT BASISU_DIR)
+    if(NOT TARGET BasisUniversal::Transcoder)
+        add_library(BasisUniversal::Transcoder INTERFACE IMPORTED)
+        set_property(TARGET BasisUniversal::Transcoder APPEND PROPERTY
+            INTERFACE_LINK_LIBRARIES basisu_transcoder)
+
+        # Retrieve include directory for FindPackageHandleStandardArgs later
+        get_target_property(BasisUniversalTranscoder_INCLUDE_DIR basisu_transcoder
+            INTERFACE_INCLUDE_DIRECTORIES)
+    endif()
+
+    if(NOT TARGET BasisUniversal::Encoder)
+        add_library(BasisUniversal::Encoder INTERFACE IMPORTED)
+        set_property(TARGET BasisUniversal::Encoder APPEND PROPERTY
+            INTERFACE_LINK_LIBRARIES basisu_encoder)
+
+        # Retrieve include directory for FindPackageHandleStandardArgs later
+        get_target_property(BasisUniversalEncoder_INCLUDE_DIR basisu_encoder
+            INTERFACE_INCLUDE_DIRECTORIES)
+    endif()
+
+# Otherwise find the source files and compile them as part of the library they
+# get linked to
+else()
+    # Disable the find root path here, it overrides the
+    # CMAKE_FIND_ROOT_PATH_MODE_INCLUDE setting potentially set in
+    # toolchains.
+    find_path(BasisUniversalEncoder_INCLUDE_DIR NAMES basisu_frontend.h HINTS
+        HINTS "${BASISU_DIR}" "${BASISU_DIR}/encoder"
+        NO_CMAKE_FIND_ROOT_PATH)
+    mark_as_advanced(BasisUniversalEncoder_INCLUDE_DIR)
+
+    find_path(BasisUniversalTranscoder_INCLUDE_DIR NAMES basisu_transcoder.h
+        HINTS "${BASISU_DIR}/transcoder" "${BASISU_DIR}"
+        NO_CMAKE_FIND_ROOT_PATH)
+    mark_as_advanced(BasisUniversalTranscoder_INCLUDE_DIR)
+
+    if(NOT TARGET BasisUniversal::Transcoder)
+        add_library(BasisUniversal::Encoder INTERFACE IMPORTED)
+        set_property(TARGET BasisUniversal::Encoder APPEND PROPERTY
+            INTERFACE_INCLUDE_DIRECTORIES ${BasisUniversalEncoder_INCLUDE_DIR})
+
+        find_path(BasisUniversalTranscoder_DIR NAMES basisu_transcoder.cpp
+            HINTS "${BASISU_DIR}/transcoder" "${BASISU_DIR}"
+            NO_CMAKE_FIND_ROOT_PATH)
+        add_library(BasisUniversalTranscoder STATIC
+            ${BasisUniversalTranscoder_DIR}/basisu_transcoder.cpp)
+        target_include_directories(BasisUniversalTranscoder PUBLIC
+            ${BasisUniversalTranscoder_INCLUDE_DIR})
+
+        if(CMAKE_CXX_COMPILER_ID MATCHES "(Apple)?(Clang)|GCC") # FIXME: What's the right pattern here?
```

Comment:
This still remains

Reply:
Same what UseCorrade.cmake does I think: https://github.com/mosra/corrade/blob/35a7b2bc29838eead10ed2d0382374402ed1a327/modules/UseCorrade.cmake#L72

Reply:
Wow, what a line, would have never figured that out correctly 😂